x-pack/filebeat/input/httpjson/input.go

```diff
@@ -222,6 +222,9 @@ func getRateLimit(header http.Header, rateLimit *RateLimit) (int64, error) {
 				if err != nil {
```

Comment:
I suggest to expand the godoc to explain what is returned (like time at which to resume in seconds since unix epoch). Then include that it can return 0 which means the rate limit has expired or no rate limit header was present.

It seems like it would be reasonable to return a `time.Time` given that is how the consumers will be using it. (Times can be zero value checked with `IsZero()`)

Reply:
Added descriptions. Returning time.Time is fine, but no necessarily simpler since we need to handle the test cases where it is easier to compare the epoch values. I will leave it as int64 for now.